### PR TITLE
Chat team

### LINKED ISSUE #4345: Show progress bar when fetching messages from mailserver
Blocked by: https://github.com/status-im/status-go/issues/1036

### User Story

As a user, I want to know when the app is fetching messages from the mailserver so I'm not wondering if the app is doing anything, or if my push notification was bogus. 

cc @andmironov for design input. My suggestion would be to have an indeterminate progress indicator near the top of the mobile app, like this (the indicator pictured is not indeterminate, but you get the gist):

image

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Feature

[comment]: # (Describe the feature you would like, or briefly summarise the bug and what you did, what you expected to happen, and what actually happens. Sections below)
*Summary*: It is confusing to receive a push notification telling me I have a new message, clicking on it and then seeing no new messages. Since the app only fetches messages after being brought to the foreground, we should show an indeterminate progress bar to inform the user that the app is working on something.

#### Expected behavior
[comment]: # (Describe what you expected to happen.)

1. I receive a Push Notification
1. I tap the PN
1. The app opens on the main screen, and a progress bar indicates it is fetching messages
1. After a few seconds, new unread messages appear
1. After a predefined time limit or if a few more seconds have gone by without no new messages appearing, the progress bar is removed

#### Actual behavior
[comment]: # (Describe what actually happened.)

1. I receive a Push Notification
1. I tap the PN
1. The app opens on the main screen, and no new unread messages are visible
1. After a few seconds, new unread messages appear

### Solution
[comment]: # (Please summarise the solution and provide a task list on what needs to be fixed.)
*Summary*: 

Right now it is not possible to know whether messages which are arriving come from the mailserver, nor if/when they have finished arriving. Therefore we have to do a best effort solution for now. For instance, we could follow these heuristics:
- Keep progress bar visible if `shhext_requestMessages` was called less than 10 seconds ago
- Keep progress bar visible if a new unread message was added less than 5 seconds ago
- Hide progress bar if `shhext_requestMessages` was called more than 30 seconds ago

### LINKED ISSUE #5371: Fetching messages gray status bar covers some part of clickable UI
### Description

*Type*: Bug

*Summary*: When the gray status bar appears it covers buttons and makes some of them not clickable. Maybe we could improve how this status bar looks like and where it appears. Note, that the bar can be clickable in some cases like "Tap to reconnect" when you lost connection with your peers.

#### Expected behavior
The status bar does not cover clickable UI. 

#### Actual behavior
The status bar prevents user from clicking "Add to contact" button or clicking on the first row in chat. 

### Reproduction

- Open Status
- Go offline (turn on Airplane mode) and online again 

### Additional Information
* Status version: 0.9.24
* Operating System: Android, iOS

### LINKED ISSUE #6440: Increase the width of mobile message bubbles.
[comment]: # (Please replace ... with your information. Remove < and >)

### User Story

As a user, I want the chat bubbles large enough to comfortably fit longer writing and use the available space most effectively.

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Improvement

[comment]: # (Describe the feature you would like, or briefly summarise the bug and what you did, what you expected to happen, and what actually happens. Sections below)
*Summary*: Improve the chat UX by better use of the screen real estate. Messages at present are too narrow and leave a lot of vacant space that could be better used to show additional content, enhancing information density especially on larger devices.

#### Expected behavior
Bubbles should be capped at 320px max width, so they can't grow any wider than that. The larger margin should be at least 64px wide and it can grow larger if necessary but not smaller than that.

#### Actual behavior
Bubbles are capped at 230px max width.

### Solution
Attached mockup with constraints

screenshot 2018-10-22 at 11 24 26

### LINKED ISSUE #7116: MVP 3 Fetching messages progress bar
# Problem

As a user, I don't want the `fetching messages` progress bar to prevent me from using touchable area such as `response to contact request` or the first item in the home items list.

# Design

https://github.com/status-im/status-react/issues/5371
https://github.com/status-im/status-react/issues/4345

# Implementation

- check the state of the designs and implementation in code 
- implement missing parts

# Acceptance Criteria

- user story is solved
- designs are implemented https://framer.cloud/MsLgV/

### PR DESCRIPTION
**Pull request by yenda**
fix #6440 increase width of mobile message bubbles on mobile

Connectivity status related issues:
fix #5371
fix #7116
fix #4345

- introduce "Connected..." status bar
- introduce fetching animation
- removes overlap of status bar with views
- add animations for status bar

(it's a long GIF!)
connectivity

screenshot_1545127749
fetching

#### Platforms (optional)
- Android
- iOS

#### Areas that maybe impacted (optional)
**Functional**
- 1-1 chats
- public chats
- group chats

### Steps to test:
- Open Status
- Check position of status bar while offline/fetching messages/connecting to mailserver (use hongkong mailservers since it sucks and will stay longer)

status: ready

### CODE REVIEW

**Comment by yenda** [comment]
@siphiuel @flexsurfer @errorists this is just a merge of bubble and status PRs no changes.
@churik can you test this one instead of the two others?

**Review by vitvly — APPROVED** [review]

**Comment by yenda** [comment]
@errorists can you check the latest build to see if you still have the issue with iphoneX?

**Comment by errorists** [comment]
@yenda no, it's gone

**Comment by yenda** [comment]
@errorists ok and what about the animations are they better now with native drivers or it's still a bit janky?

**Comment by errorists** [comment]
I think they are! one last thing and I'm done :) Can the fetching… progress bar be placed on top of the Top Bar instead of below it? Why I'm asking, it would be better if each time the progress bar is shown it didn't nudge the entire content down, my lame-ass 3d drawing of what I mean
screenshot 2018-12-19 at 12 41 48

File: src/status_im/ui/components/connectivity/subs.cljs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+                       mailserver-request-error?
+                       :t/mailserver-request-error-status
+
+                       :else nil)]
+     {:message (i18n/label (or error-label :t/connected))
+      :connected? (nil? error-label)
+      :connecting? (= error-label :t/connecting)
+      :loading-indicator? mailserver-fetching?
+      :on-press-fn #(cond
```
Comment (flexsurfer) [inline src/status_im/ui/components/connectivity/subs.cljs:44]: events logic in the subscription?  can it be moved to events?

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/ui/components/connectivity/subs.cljs (hunk 1 above)
Comment (yenda) [reply src/status_im/ui/components/connectivity/subs.cljs:44]: It's not event logic just the event that will be dispatched based on the state of the subscription

**Comment by churik** [comment]
@yenda tested on IOS 11.4 (Iphone 7). Findings:
**1) Fetching animation is overlapping by messages**
default
Video: http://take.ms/pge4g

**2) Fetching animation is not shown along with "Add to contacts"**

Steps:
- log in
- send to user contact request from another device (for "Add to contacts" to show up)
- open chat
- tap on ...
- fetch history for 24h

Expected result: can see fetching animation bar
Actual result: no fetching animation bar

**Comment by churik** [comment]
@yenda 
just after sign in after you switching to wallet and switch back to home screen welcome text "... Tap (+) button to get started" is placed on top on Android and overlapped partially by "Connecting" label, after disappearing "Connecting" label it is overlapped by status logo.
On IOS I can see empty screen. Expected that I can see this text placed on the centre of the screen as it is in current nightly.
exp
**With "Connecting" label:**
qwe

**On IOS:**
iosdf

**Comment by churik** [comment]
@yenda ping me please when it'll be ready to retest

**Comment by yenda** [comment]
@churik should be fixed now

**Comment by churik** [comment]
@yenda 
chat list is broken on IOS and Android:
img_82af01db5664-1

**Comment by yenda** [comment]
@churik yeah I fixed that in the latest build

**Comment by churik** [comment]
Tested on IOS 11.4.1 (IPhone 7) and on Android 7(LG V20)

**Pull request merged**